common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerData.kt

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.broker
+
+import android.content.Context
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants
+import com.microsoft.identity.common.java.exception.ClientException
+import lombok.ToString
+import java.util.*
+
+/**
+ * Represents Package Name and Signature Hash (hash of the app's signing certificate)
+ * of a broker app.
+ */
+@ToString
+data class BrokerData(val packageName : String,
+                      val signatureHash : String) {
+
+    override fun equals(other: Any?): Boolean {
+        if (other !is BrokerData){
+            return false
+        }
+
+        return packageName.equals(other.packageName, ignoreCase = true) &&
+                signatureHash == other.signatureHash
+    }
+
+    override fun hashCode(): Int {
+        var result = packageName.hashCode()
+        result = 31 * result + signatureHash.hashCode()
```

Comment:
why 31?

Reply:
This was actually autogenerated by the IDE.

As for your question - https://www.geeksforgeeks.org/why-does-javas-hashcode-in-string-use-31-as-a-multiplier/